hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyDefault.java

```diff
@@ -1236,11 +1245,17 @@ public DatanodeStorageInfo chooseReplicaToDelete(
         minSpace = free;
         minSpaceStorage = storage;
       }
+      if (considerLoad2chooseReplicaDeleting && nodeLoad < minLoad) {
+        minLoad = nodeLoad;
+        minLoadStorage = storage;
+      }
     }
 
     final DatanodeStorageInfo storage;
     if (oldestHeartbeatStorage != null) {
```

Comment:
this `if else` code section seems problematic: After exiting from the previous for loop, minSpaceStorage should not be null because minSpace is initialized to Long.MAX_VALUE. So, `storage = minSpaceStorage;` will always be executed.

Reply:
@hfutatzhanghb, have you confirmed whether this is a bug or no? If no, could you share why this is not a bug? Thanks,